apps/linear_algebra/benchmarks/macros.h

```diff
@@ -1,19 +1,23 @@
 #define time_it(code)                               \
     double elapsed = 0;                             \
-    for (int iters = 20; ; iters *= 2) {            \
-        double start = current_time();              \
-        for (int i = 0; i < iters; i++) {           \
-            code;                                   \
-        }                                           \
-        double end = current_time();                \
-        elapsed = 1000 * (end - start);             \
-        if (elapsed > 100000) {                     \
-            elapsed /= iters;                       \
-            break;                                  \
-        }                                           \
+    for (int iters = 20; ; iters *= 2) {                    \
```

Comment:
Maybe not for this PR... but how about using the benchmark() function from apps/support/benchmark.h?